ReactAndroid/src/main/java/com/facebook/react/ReactActivity.java

```diff
@@ -86,7 +86,7 @@ protected ReactNativeHost getReactNativeHost() {
    * determining this (default just checks {@code BuildConfig}).
    */
   protected boolean getUseDeveloperSupport() {
-    return ((ReactApplication) getApplication()).getReactNativeHost().getUseDeveloperSupport();
+    return getReactNativeHost().getUseDeveloperSupport();
```

Comment:
@satya164 I submit a new commit which added UIExplorerApplication and MoviesApplication, please review it. Thanks